homeassistant/components/suez_water/coordinator.py

```diff
@@ -0,0 +1,113 @@
+"""Suez water update coordinator."""
+
+import asyncio
+from datetime import datetime, timedelta
+from typing import Any
+
+from pysuez import SuezClient
+from pysuez.client import PySuezError
+
+from homeassistant.core import _LOGGER, HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+
+class AggregatedSensorData:
+    """Hold suez water aggregated sensor data."""
+
+    def __init__(
+        self,
+        value: float,
+        currentMonth: Any,
+        previousMonth: Any,
+        previousYear: Any,
+        currentYear: Any,
+        history: Any,
+        highestMonthlyConsumption: Any,
+        attribution: str,
+    ) -> None:
+        """Create aggregated sensor data."""
+        self.value = value
+        self.currentMonth = currentMonth
+        self.previousMonth = previousMonth
+        self.previousYear = previousYear
+        self.currentYear = currentYear
+        self.history = history
+        self.highestMonthlyConsumption = highestMonthlyConsumption
+        self.attribution = attribution
+
+
+class SuezWaterCoordinator(DataUpdateCoordinator):
```

Comment:
```suggestion
class SuezWaterCoordinator(DataUpdateCoordinator[..]):
```
Be sure to type the return value of `_async_update_data` as the generic of the coordinator to automatically type `.data`